rag_experiment_accelerator/evaluation/eval.py

```diff
@@ -546,68 +541,75 @@ def compute_metrics(
     """
-    if metric_type == "lcsstr":
-        score = lcsstr(actual, expected)
-    elif metric_type == "lcsseq":
-        score = lcsseq(actual, expected)
-    elif metric_type == "cosine":
-        score = cosine(actual, expected)
-    elif metric_type == "jaro_winkler":
-        score = jaro_winkler(actual, expected)
-    elif metric_type == "hamming":
-        score = hamming(actual, expected)
-    elif metric_type == "jaccard":
-        score = jaccard(actual, expected)
-    elif metric_type == "levenshtein":
-        score = levenshtein(actual, expected)
-    elif metric_type == "fuzzy":
-        score = fuzzy(actual, expected)
-    elif metric_type == "bert_all_MiniLM_L6_v2":
-        all_MiniLM_L6_v2 = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2")
-        score = compare_semantic_document_values(actual, expected, all_MiniLM_L6_v2)
-    elif metric_type == "bert_base_nli_mean_tokens":
-        base_nli_mean_tokens = SentenceTransformer(
-            "sentence-transformers/bert-base-nli-mean-tokens"
-        )
-        score = compare_semantic_document_values(actual, expected, base_nli_mean_tokens)
-    elif metric_type == "bert_large_nli_mean_tokens":
-        large_nli_mean_tokens = SentenceTransformer(
-            "sentence-transformers/bert-large-nli-mean-tokens"
-        )
-        score = compare_semantic_document_values(
-            actual, expected, large_nli_mean_tokens
-        )
-    elif metric_type == "bert_large_nli_stsb_mean_tokens":
-        large_nli_stsb_mean_tokens = SentenceTransformer(
-            "sentence-transformers/bert-large-nli-stsb-mean-tokens"
-        )
-        score = compare_semantic_document_values(
-            actual, expected, large_nli_stsb_mean_tokens
-        )
-    elif metric_type == "bert_distilbert_base_nli_stsb_mean_tokens":
-        distilbert_base_nli_stsb_mean_tokens = SentenceTransformer(
-            "sentence-transformers/distilbert-base-nli-stsb-mean-tokens"
-        )
-        score = compare_semantic_document_values(
-            actual, expected, distilbert_base_nli_stsb_mean_tokens
-        )
-    elif metric_type == "bert_paraphrase_multilingual_MiniLM_L12_v2":
-        paraphrase_multilingual_MiniLM_L12_v2 = SentenceTransformer(
-            "sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2"
-        )
-        score = compare_semantic_document_values(
-            actual, expected, paraphrase_multilingual_MiniLM_L12_v2
-        )
-    elif metric_type == "llm_answer_relevance":
-        score = llm_answer_relevance(response_generator, question, actual)
-    elif metric_type == "llm_context_precision":
-        score = llm_context_precision(response_generator, question, retrieved_contexts)
-    elif metric_type == "llm_context_recall":
-        score = llm_context_recall(
-            response_generator, question, expected, retrieved_contexts
-        )
-    else:
-        pass
+    match metric_type:
+        case "lcsstr":
```

Comment:
Look, these constants look weird. This whole switch case can be replaced with the following structure
`
import importlib
def dynamic_import(module_name, function_name):
    try:
        module = importlib.import_module(module_name)
        return getattr(module, function_name)
    except (ImportError, AttributeError) as e:
        some error handling, probably ignore in case of metrics
`

And move these metric functions into a separate file, for example evaluations/metrics.py
So then you can call

`
metric_func = dynamic_import("rag_experiment_accelerator.evaluation.metrics", metric_type)
score = metric)func(actual, expected)
`